barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -3528,8 +3528,15 @@ std::vector<Row> AvmTraceBuilder::finalize()
     for (size_t i = 0; i < main_trace_size; i++) {
         main_trace[i].main_sel_execution_row = FF(1);
     }
-    // This selector corresponds to the last row of the EXECUTION trace.
-    main_trace.back().main_sel_last = FF(1);
+
+    // Toggle start of the execution row
+    if (main_trace_size > 0) {
+        main_trace[0].main_sel_start_exec = FF(1);
+    }
```

Comment:
Is this actually row 0 or row 1? Do we add the extra first "shift" row afterwards? Maybe a comment about that.

Reply:
Yes, we add the extra shift row later.